src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/AccessTokenFactory.php

```diff
@@ -0,0 +1,129 @@
+                ->beforeNormalization()
+                    ->ifString()
+                    ->then(static function (string $v): array { return [$v]; })
+                ->end()
+                ->cannotBeEmpty()
+                ->defaultValue([
+                    'security.access_token_extractor.header',
+                ])
+                ->scalarPrototype()->end()
+            ->end()
+        ;
+    }
+
+    public function getPriority(): int
+    {
+        return self::PRIORITY;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getKey(): string
+    {
+        return 'access_token';
+    }
+
+    public function createAuthenticator(ContainerBuilder $container, string $firewallName, array $config, string $userProviderId): string
+    {
+        $userProvider = new Reference($config['user_provider'] ?? $userProviderId);
+        $successHandler = isset($config['success_handler']) ? new Reference($this->createAuthenticationSuccessHandler($container, $firewallName, $config)) : null;
+        $failureHandler = isset($config['failure_handler']) ? new Reference($this->createAuthenticationFailureHandler($container, $firewallName, $config)) : null;
+        $authenticatorId = sprintf('security.authenticator.access_token.%s', $firewallName);
+        $extractorId = $this->createExtractor($container, $firewallName, $config['token_extractors']);
+
+        $container
+            ->setDefinition($authenticatorId, new ChildDefinition('security.authenticator.access_token'))
+            ->replaceArgument(0, $userProvider)
+            ->replaceArgument(1, new Reference($config['token_handler']))
+            ->replaceArgument(2, new Reference($extractorId))
+            ->replaceArgument(3, $successHandler)
+            ->replaceArgument(4, $failureHandler)
+            ->replaceArgument(5, $config['realm'])
+        ;
+
+        return $authenticatorId;
+    }
+
+    /**
+     * @param array<string> $extractors
+     */
+    private function createExtractor(ContainerBuilder $container, string $firewallName, array $extractors): string
+    {
+        $aliases = $this->getExtractorAliases();
+        $extractors = array_map(static function (string $extractor) use ($aliases): string {
+            return $aliases[$extractor] ?? $extractor;
+        }, $extractors);
+
+        if (count($extractors) === 1) {
+            return current($extractors);
+        }
+        $extractorId = sprintf('security.authenticator.access_token.chain_extractor.%s', $firewallName);
+        $container
+            ->setDefinition($extractorId, new ChildDefinition('security.authenticator.access_token.chain_extractor'))
+            ->replaceArgument(0, array_map(function (string $extractorId): Reference {return new Reference($extractorId);}, $extractors))
+        ;
+
+        return $extractorId;
+    }
+
+    /**
+     * @return array<string, string>
+     */
+    private function getExtractorAliases(): array
```

Comment:
Is this method really useful? It can be inlined IMHO

Reply:
Yes it can be inlined. This is just that I prefer when things are split.
I can change this part if erveryone agrees with that

Reply:
Agreed, please inline it :)